pkg/binding/interfaces.go

```diff
@@ -84,13 +86,26 @@ var ErrNotStructured = errors.New("message is not in structured mode")
 // ErrNotBinary returned by Message.Binary for non-binary messages.
 var ErrNotBinary = errors.New("message is not in binary mode")
 
+// MessagePayload allows to read a message payload or as a byte array or as a reader
+// Message implementers must re
+type MessagePayloadReader interface {
```

Comment:
I think this is an undue burden on the Message implementer. If they provide an io.Reader then io.Copy to a bytes.Buffer is maximally efficient, and the streaming case is supported. io.Reader is a standard Go idiom (net.http and many other standard libs) so I don't think we should hide it in a non-standard interface.